### ISSUE TITLE
Unable to reproduce training results

### ISSUE THREAD

**Issue by Hong-Bo** [makeuptransfer/SCGAN#10]
First, great work and thanks for providing training codes. 

Out of curiosity, I tried to retrain the model and here's what I've done to reproduce your work:
As a start, I cloned the repository and set up the required environments (python 3.8, torch 1.6, torchvision 0.7). 
Then downloaded the training dataset and parsed all the original segmentation images using `scripts/handle_parsing.py', (makeup and non-makeup subdirectories). 
The weights file, vgg_conv.pth, was downloaded afterwards and put to the right place.
Finally, the training was initiated with the recommended training command line. 

However, after one night's training, the training results seemed very furry to me. Here's some examples
![13_0](https://user-images.githubusercontent.com/25097553/126925759-a83e92e1-bd61-481e-a6dd-f4fbc17860a7.jpg)
![14_0](https://user-images.githubusercontent.com/25097553/126925762-3c44f11f-73dc-4be9-b73e-e59aea757fe5.jpg)
![15_0](https://user-images.githubusercontent.com/25097553/126925774-b7b40630-f2a7-4e84-b886-f3022d976ad3.jpg)
![30_0](https://user-images.githubusercontent.com/25097553/126925780-fcfd94ea-ebe7-4f45-b945-6bef6c64eeaf.jpg)
They were saved at 13, 14, 15, 30 epochs respectively.

A deep analysis is greatly appreciated. Thank you.

**Comment by makeuptransfer** [comment]
Because the training images are randomly selected, so a different order will have a different impact on the results. From your training results, it seems as if the role of makeup loss is not strong enough. You can try to adjust the weight of the loss

**Comment by Yueming6568** [comment]
I also have this problem and many people meet it. I think it is a bug, can you help us solve it ?

**Comment by Hong-Bo** [comment]
> I also have this problem and many people meet it. I think it is a bug, can you help us solve it ?

Shortly after I found this problem, I read the codes thoroughly and realized the implementation of the codes differ slightly with what had been done in the paper.  But I am not sure how many places are different.

**Comment by Yueming6568** [comment]
> > I also have this problem and many people meet it. I think it is a bug, can you help us solve it ?
> 
> Shortly after I found this problem, I read the codes thoroughly and realized the implementation of the codes differ slightly with what had been done in the paper. But I am not sure how many places are different.

so,  do you solve it?
i think the difference is not crucial

**Comment by Hong-Bo** [comment]
> > > I also have this problem and many people meet it. I think it is a bug, can you help us solve it ?
> > 
> > 
> > Shortly after I found this problem, I read the codes thoroughly and realized the implementation of the codes differ slightly with what had been done in the paper. But I am not sure how many places are different.
> 
> so, do you solve it?
> i think the difference is not crucial

I have been working on it since this morning. I'll let you know if there's some progress asap, if you're interested.

**Comment by Yueming6568** [comment]
> > > > I also have this problem and many people meet it. I think it is a bug, can you help us solve it ?
> > > 
> > > 
> > > Shortly after I found this problem, I read the codes thoroughly and realized the implementation of the codes differ slightly with what had been done in the paper. But I am not sure how many places are different.
> > 
> > 
> > so, do you solve it?
> > i think the difference is not crucial
> 
> I have been working on it since this morning. I'll let you know if there's some progress asap, if you're interested.

thank you very much! waiting for your good news!

**Comment by makeuptransfer** [comment]
The model is consistent with the description of the paper. the first training result is good, so I didn't pay too much attention to the traning. Thanks for your questions! i will check it

**Comment by Hong-Bo** [comment]
> The model is consistent with the description of the paper. the first training result is good, so I didn't pay too much attention to the traning. Thanks for your questions! i will check it

One of the aforementioned differences is that the local perceptual loss is not added to the total losses of generator. Adding it to the total losses makes training result better. Grateful to your reply and wait for your results of scrutinization.

**Comment by semi-supervised-paper** [comment]
Any update here?

**Comment by Hong-Bo** [comment]
> The model is consistent with the description of the paper. the first training result is good, so I didn't pay too much attention to the traning. Thanks for your questions! i will check it

Any progress?
